tests/unit/components/form/GitForm.spec.js

```diff
@@ -0,0 +1,74 @@
+import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-jest';
+import { shallowMount } from '@vue/test-utils';
+import GitForm from 'src/components/form/GitForm.vue';
+import { getProjectById, saveProject } from 'src/composables/Project';
+
+installQuasarPlugin();
+
+describe('Test component: GitForm', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    saveProject({
+      id: 'test',
+    });
+    wrapper = shallowMount(GitForm, {
+      props: {
+        projectName: 'test',
+      },
+    });
+  });
+
+  describe('Test variables initialization', () => {
+    describe('Test props: projectName', () => {
+      it('should match "model"', () => {
+        expect(wrapper.vm.props.projectName).toEqual('test');
+      });
+    });
+
+    describe('Test variable: repository', () => {
+      it('should be undefined on no git configuration in project', () => {
+        expect(wrapper.vm.repository).toBeUndefined();
+      });
+
+      it('should be not null on git configuration in project', () => {
+        saveProject({ id: 'test', git: { repository: 'testRepository' } });
+        wrapper = shallowMount(GitForm, {
+          props: {
+            projectName: 'test',
+          },
+        });
+        expect(wrapper.vm.repository).toEqual('testRepository');
+      });
+    });
+
+    describe('Test variable: token', () => {
+      it('should be undefined on no git configuration in project', () => {
+        expect(wrapper.vm.token).toBeUndefined();
+      });
+
+      it('should be not null on git configuration in project', () => {
+        saveProject({ id: 'test', git: { token: 'testToken' } });
+        wrapper = shallowMount(GitForm, {
+          props: {
+            projectName: 'test',
+          },
+        });
+        expect(wrapper.vm.token).toEqual('testToken');
+      });
+    });
+  });
+
+  describe('Test function: onSubmit', () => {
+    it('should emit an event', () => {
```

Comment:
Here, the emission of the event is not checked. Only the modification of the localStorage is.

Reply:
done